### ISSUE TITLE
Clean up flaky tests in `test_inference.py`

### ISSUE THREAD

**Issue by natestemen** [unitaryfoundation/mitiq#1509]
We seem to have a few flaky tests which have popped up recently giving the dreaded red X. Since our test often take 20-30min to run, restarting them due to a flaky test is undesirable.

- In https://github.com/unitaryfund/mitiq/pull/1492 we saw [`mitiq/zne/tests/test_inference.py::test_plot_data_exp_factory[ExpFactory]`](https://github.com/unitaryfund/mitiq/blob/a48f35f9ba7fab2f491c2baac2a905b6f3706b40/mitiq/zne/tests/test_inference.py#L887-L902) fail.
- In https://github.com/unitaryfund/mitiq/actions/runs/3130210903/jobs/5080201273 we see [`mitiq/zne/tests/test_inference.py::test_plot_fit[LinearFactory]`](https://github.com/unitaryfund/mitiq/blob/55653c73bfa20a796eff8abb6d0eec2ad881174e/mitiq/zne/tests/test_inference.py#L913) fail.

**Comment by natestemen** [comment]
[master](https://github.com/unitaryfund/mitiq/actions/runs/3174423778/jobs/5171198696) just failed on `mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory]`

**Comment by natestemen** [comment]
[another](https://github.com/unitaryfund/mitiq/actions/runs/3276503155/jobs/5392631981) fail on `mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory]`

**Comment by natestemen** [comment]
[another](https://github.com/unitaryfund/mitiq/actions/runs/3321940639/jobs/5490277619) fail on `mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory]`

**Comment by natestemen** [comment]
[another](https://github.com/unitaryfund/mitiq/actions/runs/3358815258/jobs/5566225208) fail with both
- `mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory]`
- `mitiq/zne/tests/test_inference.py::test_plot_data[RichardsonFactory]`

**Comment by natestemen** [comment]
[another](https://github.com/unitaryfund/mitiq/actions/runs/3363585326/jobs/5576920514) fail on `mitiq/zne/tests/test_inference.py::test_plot_fit[LinearFactory]`

**Comment by natestemen** [comment]
[another](https://github.com/unitaryfund/mitiq/actions/runs/3387624289/jobs/5628609005) fail on `mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory]`

**Comment by natestemen** [comment]
All of the errors linked above have the same pattern, and it's due to `pytest-xdist` workers going down while running the test.
```
mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory] 
Current thread 0x0000700007798000 (most recent call first):
[gw2] node down: Not properly terminated
...
[gw2] [ 89%] FAILED mitiq/zne/tests/test_inference.py::test_plot_data[LinearFactory] 

replacing crashed worker gw2
```

This seems to be a known issue in [`pytest-xdist`](https://github.com/pytest-dev/pytest-xdist) which we use to speed up test runtime by distributing tests across multiple CPUs. The two issues I've found that are related to this are

1. https://github.com/pytest-dev/pytest-xdist/issues/466, and
2. https://github.com/pytest-dev/pytest-xdist/issues/714.

The problem should go away if we removed distribution of the tests, but this IMO is not a reasonable trade-off given the speed-up it provides over running the test sequentially. Ideally `pytest-xdist` would be able to recover from this situation and re-run the test that was running during the worker failure. For that reason I recommend we keep this open for now, but take no action. That means rerunning failures when we see this, but monitoring it in case a fix is implemented upstream, or the issues arise more/less often and we decide to take action.

WDYT @Misty-W @andreamari

**Comment by Misty-W** [comment]
I agree to keep this open for now and monitor once or twice per milestone cycle. I'll go ahead and remove it from Milestone 0.21 because it's unlikely to be solved by the time it closes. If/when we see a solution that needs action from our side, we can add it to whichever milestone that ends up being.

**Comment by andreamari** [comment]
Thanks @natestemen! I agree with you and @Misty-W: no action is probably the best strategy right now.

**Comment by natestemen** [comment]
Even though the two pytest issues linked above are still open this seems to have resolved itself. Perhaps some numpy upgrade meant more numerical stability.
